examples/step-6/doc/intro.dox

```diff
@@ -103,10 +103,42 @@ up with these &ldquo;hanging nodes&rdquo; if we do this.
 
 <h3> How to deal with hanging nodes </h3>
 
+The methods using triangular meshes mentioned above go to great
+lengths to make sure that each vertex is a vertex of all adjacent
+cells -- i.e., that there are no hanging nodes. This then
+automatically makes sure that we can define shape functions in such a
+way that they are globally continuous (if we use the common $Q_p$
+Lagrange finite element methods we have been using so far in the
+tutorial programs, as represented by the FE_Q class).
+
+On the other hand, if we define shape functions on meshes with hanging
+nodes, we may end up with shape functions that are not continuous. To
+see this, think about the situation above where the top right cell is
+not refined, and consider for a moment the use of a bilinear finite
+element. In that case, the shape functions associated with the hanging
+nodes are defined in the obvious way on the two small cells adjacent
+to each of the hanging nodes. But how do we extend them to the big
+adjacent cells? Clearly, the function's extension to the big cell
+cannot be bilinear because then it needs to be linear along each edge
+of the large cell, and that means that it needs to be zero on the
+entire edge because it needs to be zero on the two vertices of the
+large cell on that edge. But it is not zero at the hanging node itself
+when seen from the small cells' side -- so it is not continuous.
```

Comment:
in this case a picture would help, IMO. You can plot a single shape function (on vertex shared by coarse and refined cell) without distributing constraints.